src/basic/socket-util.c

```diff
@@ -1171,6 +1171,18 @@ struct cmsghdr* cmsg_find(struct msghdr *mh, int level, int type, socklen_t leng
         return NULL;
 }
 
+void* cmsg_find_and_copy_data(struct msghdr *mh, int level, int type, void *buf, size_t buf_len) {
+        struct cmsghdr *cmsg;
+
+        assert(mh);
+
+        cmsg = cmsg_find(mh, level, type, buf_len == SIZE_MAX ? (socklen_t) -1 : CMSG_LEN(buf_len));
+        if (!cmsg)
+                return NULL;
+
+        return memcpy_safe(buf, CMSG_DATA(cmsg), buf_len == SIZE_MAX ? cmsg->cmsg_len : buf_len);
```

Comment:
please add a comment here explaining why this is necessary - quoting the manpage is enough

Reply:
still unaddressed?

Reply:
Do you think so? What kind of message do you want ?

Reply:
I meant to add it as a code comment, as it's obscure enough so it's bound to look puzzling in the future when one has to look here, not just as a commit message. A short version of the commit message would be more than enough.

Reply:
OK. I will do.